modules/Bio/EnsEMBL/Hive/Worker.pm

```diff
@@ -255,7 +255,7 @@ sub current_role {
         }
         my $new_role = shift @_;
         if( my $to_analysis = $new_role && $new_role->analysis ) {
-            $self->worker_say( "specializing to ".$to_analysis->logic_name.'('.$to_analysis->dbID.')' );
+            $self->worker_say( "specializing to ".$to_analysis->logic_name.'('.($to_analysis->dbID // 'unstored').')' );
```

Comment:
Seeing as this line is already being modified, could you add more whitespace to it so that it is easier to read? It might also make sense to split it into two or more, even without whitespace it is already on the long side.